docs/site/Deploying-to-IBM-Cloud.md

```diff
@@ -0,0 +1,211 @@
+---
+lang: en
+title: 'Deploying to IBM Cloud'
+keywords: LoopBack
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Deploying-to-IBM-Cloud.html
+---
+
+This is a guide for deploying LoopBack 4 (LB4) apps to IBM Cloud. In the setup
+explained below, your app will use a local instance of Cloundant when running
+locally, and a provisioned Cloudant service when running on the cloud.
+
+Before we begin make sure the following are installed.
+
+1.  [Node.js](https://nodejs.org/en/download/) >= 8.9.0.
+2.  [LB4 CLI](https://www.npmjs.com/package/@loopback/cli) (`lb4` command).
+3.  [Cloud Foundy CLI](https://docs.cloudfoundry.org/cf-cli/install-go-cli.html).
+4.  [Docker](https://www.docker.com/).
+
+Consider the
+["todo" example](https://github.com/strongloop/loopback-next/tree/master/examples/todo)
+from the [loopback-next repository](https://github.com/strongloop/loopback-next)
+as a basis for the instruction.
+
+You can quickly clone the "todo" example app by running the command:
+
+```sh
+$ lb4 example todo
+```
+
+## Local setup
+
+### Database
+
+1.  [Download Cloudant image](https://hub.docker.com/r/ibmcom/cloudant-developer/).
+2.  Ensure port 8080 is free and start Cloudant:
+
+```
+docker run \
+      --volume cloudant:/srv \
+      --name cloudant-developer \
+      --publish 8080:80 \
+      --hostname cloudant.dev \
+      ibmcom/cloudant-developer
+```
+
+3.  Log on to the local Cloudant Dashboard using `admin`:`pass` as the
+    crendentials.
+4.  Create a database named `todo`.
+
+### Updates to the app
+
+#### Updating datasource
+
+Update `db.datasource.json` to use the Cloudant connector.
+
+```js
+{
+  "name": "db",
+  "connector": "cloudant",
+  "url": "http://admin:pass@localhost:8080",
+  "database": "todo",
+  "modelIndex": ""
+}
+```
+
+Install the `loopback-connector-cloudant` package.
+
+```sh
+$ npm i loopback-connector-cloudant -s
+```
+
+#### Updating npm script
+
+Remove the `prestart` script from `package.json`, since we don't want to do any
+building on the cloud.
+
+#### Updating application
+
+We will use the `cfenv` module to simplify some of the Cloud Foundry related
+operations. Install `cfenv` in the project directory.
+
+```sh
+$ npm i cfenv -s
+```
+
+Update the `index.ts` file to the following to enable service binding.
+
+```ts
+import {TodoListApplication} from './application';
```

Comment:
I got errors when somewhat copying and pasting your code snippet.  The 2 errors:
1.  In the line of ` app.bind('datasources.config.db').to(datasourceDb);`, `app` hasn't been defined yet
2. `const datasourceDb` meaning it's not changeable but indeed we are assigning another value a few lines after. 

For your reference, here is my index.ts:
```

import {ShoppingApplication} from './application';
import {ApplicationConfig} from '@loopback/core';

// IBM Cloud deployment
let datasourceDb = require('./datasources/user.datasource.json');
const cfenv = require('cfenv');
const appEnv = cfenv.getAppEnv();

export {ShoppingApplication};

export async function main(options?: ApplicationConfig) {
  // Set the port assigned for the app
  if (!options) options = {};
  if (!options.rest) options.rest = {};

  const app = new ShoppingApplication(options);
  if (appEnv.isLocal) options.rest.port = options.rest.port;
  else {
    datasourceDb = Object.assign({}, datasourceDb, {
      url: appEnv.getServiceURL('myCloudant'),
    });
    app.bind('datasources.config.db').to(datasourceDb);
    options.rest.port = appEnv.port;
  }

  await app.boot();
  await app.start();

  const url = app.restServer.url;
  console.log(`Server is running at ${url}`);
  console.log(`Try ${url}/ping`);

  return app;
}
```

Reply:
Fixed in the latest commit. Thanks.